templates/wechatgame/first-screen.js

```diff
@@ -173,40 +302,76 @@ function updateTexture() {
+    gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_WRAP_T, gl.CLAMP_TO_EDGE);
+    gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, gl.RGBA, gl.UNSIGNED_BYTE, bg);
+}
+
+function drawTexture(gl, program, texture, vertexBuffer, vertexFormatLength) {
     gl.useProgram(program);
     gl.activeTexture(gl.TEXTURE0);
     gl.bindTexture(gl.TEXTURE_2D, texture);
     var uSampler = gl.getUniformLocation(program, 'u_Sampler');
     gl.uniform1i(uSampler, 0);
     gl.bindBuffer(gl.ARRAY_BUFFER, vertexBuffer);
-    var vertexFormatLength = 4;
     var aPosition = gl.getAttribLocation(program, 'a_Position');
     gl.enableVertexAttribArray(aPosition);
     gl.vertexAttribPointer(aPosition, 2, gl.FLOAT, false, vertexFormatLength * 4, 0);
     var aTexCoord = gl.getAttribLocation(program, 'a_TexCoord');
     gl.enableVertexAttribArray(aTexCoord);
     gl.vertexAttribPointer(aTexCoord, 2, gl.FLOAT, false, vertexFormatLength * 4, vertexFormatLength * 2);
     gl.drawArrays(gl.TRIANGLE_STRIP, 0, 4);
-    gl.useProgram(programProgress);
-    var uCurrentProgress = gl.getUniformLocation(programProgress, 'u_CurrentProgress');
+}
+
+function drawProgressBar(gl, program, vertexBuffer, vertexFormatLength, progress, progressBarColor, progressBackground) {
+    gl.useProgram(program);
+    var uCurrentProgress = gl.getUniformLocation(program, 'u_CurrentProgress');
     gl.uniform1f(uCurrentProgress, progress);
-    var uProgressBarColor = gl.getUniformLocation(programProgress, 'u_ProgressBarColor');
+    var uProgressBarColor = gl.getUniformLocation(program, 'u_ProgressBarColor');
     gl.uniform4fv(uProgressBarColor, progressBarColor);
-    var uProgressBackground = gl.getUniformLocation(programProgress, 'u_ProgressBackground');
+    var uProgressBackground = gl.getUniformLocation(program, 'u_ProgressBackground');
     gl.uniform4fv(uProgressBackground, progressBackground);
-    gl.bindBuffer(gl.ARRAY_BUFFER, vertexBufferProgress);
-    aPosition = gl.getAttribLocation(programProgress, 'a_Position');
+    gl.bindBuffer(gl.ARRAY_BUFFER, vertexBuffer);
+    var aPosition = gl.getAttribLocation(program, 'a_Position');
     gl.enableVertexAttribArray(aPosition);
+    var vertexFormatLength = 4;
     gl.vertexAttribPointer(aPosition, 2, gl.FLOAT, false, vertexFormatLength * 3, 0);
-    var aProgress = gl.getAttribLocation(programProgress, 'a_Progress');
+    var aProgress = gl.getAttribLocation(program, 'a_Progress');
     gl.enableVertexAttribArray(aProgress);
     gl.vertexAttribPointer(aProgress, 1, gl.FLOAT, false, vertexFormatLength * 3, vertexFormatLength * 2);
     gl.drawArrays(gl.TRIANGLE_STRIP, 0, 4);
 }
 
+function draw() {
+    gl.enable(gl.BLEND);
+    gl.blendFunc(gl.SRC_ALPHA, gl.ONE_MINUS_SRC_ALPHA);
+    gl.clearColor(0.0, 0.0, 0.0, 0.0);
+    gl.clear(gl.COLOR_BUFFER_BIT);
+    // draw background
+    drawTexture(gl, programBg, bgTexture, bgVertexBuffer, 4);
+    // draw logo
+    drawTexture(gl, program, texture, vertexBuffer, 4);
```

Comment:
It is better to rename texture to logoTexture.

Reply:
logo, background and slogon use the simialar function. So I named it `drawTexture`, which is more general.